int64 quad_current
int64 reach_current